packages/odyssey/src/scss/components/_radio-button.scss

```diff
@@ -64,7 +64,8 @@
     }
   }
 
-  &[data-invalid] {
+  &:invalid,
```

Comment:
is the application of this pseudo class something we have any control over within the HTML/JS or is it entirely driven by browser behavior?

Reply:
Built-in validation has defaults that can be overridden by attrs like `pattern`. On the JS side, we can use the constraint validation API (or a library) to modify the browser's behavior (or validate non-HTML components). More info here: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#validating_forms_using_javascript